src/blockcrawler/core/types.py

```diff
@@ -98,6 +98,36 @@ def __rsub__(self, other):
         else:
             return NotImplemented
 
+    def __mul__(self, other):
+        if isinstance(other, self.__class__):
+            return HexInt(self.int_value * other.int_value)
+        elif isinstance(other, int):
+            return HexInt(self.int_value * other)
+        else:
+            return NotImplemented
+
+    def __rmul__(self, other):
+        return self.__mul__(other).int_value
+
+    def __truediv__(self, other):
+        if isinstance(other, self.__class__):
+            return HexInt(self.int_value // other.int_value)
+        elif isinstance(other, int):
+            return HexInt(self.int_value // other)
+        else:
+            return NotImplemented
+
+    def __rtruediv__(self, other):
+        if isinstance(other, self.__class__):
+            return HexInt(other.int_value // self.int_value)
+        elif isinstance(other, int):
+            return HexInt(other // self.int_value).int_value
```

Comment:
There is no reason to convert to HexInt and back to int, as it's already an int based on the check.

Reply:
Addressed.